### PR TITLE
[EPMEDU-2511]: Disable phone number prefix selection for prod

### PR DESCRIPTION
**Pull request by Fileek**
[Jira ticket](https://jira.epam.com/jira/browse/EPMEDU-2511)

  - Added `isCountrySelectorClickable` to `ProfileState` and `EnterPhoneState`
  - Set `isCountrySelectorClickable` to `false` when flavor is prod

| Dev  | Prod |
| ------------- | ------------- |
| <video src="https://github.com/AnimealProject/animeal_android/assets/83027107/1160bf69-6efe-43b0-a206-11ffd66ce82b"> | <video src="https://github.com/AnimealProject/animeal_android/assets/83027107/2197aa06-3f5b-4f48-9888-6f22696b8580"> |

### CODE REVIEW

File: shared/feature/profile/src/main/java/com/epmedu/animeal/profile/presentation/ui/ProfileInputForm.kt [hunk 1]
```diff
@@ -92,7 +92,7 @@ private fun ProfileState.PhoneInput(
         onValueChange = { onEvent(PhoneNumberChanged(it)) },
         error = phoneNumberError.asString(),
         isEnabled = isPhoneNumberEnabled,
-        isFlagClickable = isPhoneNumberEnabled,
+        isFlagClickable = isFlagClickable,
```
Comment (ozkarram) [inline shared/feature/profile/src/main/java/com/epmedu/animeal/profile/presentation/ui/ProfileInputForm.kt]: `isCountrySelectorClickable`?

File: app/src/main/java/com/epmedu/animeal/component/BuildConfigProviderImpl.kt [hunk 2]
```diff
@@ -9,4 +9,9 @@ internal class BuildConfigProviderImpl : BuildConfigProvider {
     override val mapBoxPublicKey: String = BuildConfig.MAPBOX_PUBLIC_TOKEN
     override val mapBoxStyleURI: String = BuildConfig.MAPBOX_STYLE_URI
     override val isDebug: Boolean = BuildConfig.DEBUG
+    override val isProd: Boolean = BuildConfig.FLAVOR == PROD_FLAVOR
```
Comment (ozkarram) [inline app/src/main/java/com/epmedu/animeal/component/BuildConfigProviderImpl.kt]: isProdFlavor?

**Review by ozkarram — CHANGES_REQUESTED** [review]

File: shared/feature/profile/src/main/java/com/epmedu/animeal/profile/presentation/ui/ProfileInputForm.kt (hunk 1 above)
Comment (Fileek) [reply shared/feature/profile/src/main/java/com/epmedu/animeal/profile/presentation/ui/ProfileInputForm.kt]: Renamed

File: app/src/main/java/com/epmedu/animeal/component/BuildConfigProviderImpl.kt (hunk 2 above)
Comment (Fileek) [reply app/src/main/java/com/epmedu/animeal/component/BuildConfigProviderImpl.kt]: Renamed

**Review by ozkarram — APPROVED** [review]

**Pull request merged**